src/hip_memory.cpp

```diff
@@ -1637,7 +1637,15 @@ hipError_t ihipMemcpy2D(void* dst, size_t dpitch, const void* src, size_t spitch
                     stream->locked_copySync((unsigned char*)dst + i * dpitch,
                                     (unsigned char*)src + i * spitch, width, kind);
             } else {
-                stream->locked_copy2DSync(dst, src, width, height, spitch, dpitch, kind);
+                int HIP_MEMCPY2D_FORCE_SDMA = 0;
+                ihipReadEnv_I(&HIP_MEMCPY2D_FORCE_SDMA, "HIP_MEMCPY2D_FORCE_SDMA", "HIP_MEMCPY2D_FORCE_SDMA",
```

Comment:
Why are we reading this env variable in every memcpy2D? Is it a requirement for the user to be able to turn this on or off while the process is still running? We should probably handle it like the rest of the env variables that are read once during init.
Also we seem to be changing the default from SDMA enabled to SDMA disabled.

Reply:
Yes your comments are valid though I created this PR just to help checking and unblocking #878 . If we keep this as long term change then I will update this before you merge it. Let us wait for 2.1 test results on stability of SDMA. If SDMA issues persist then we can keep default path using 2D kernel for now.